### PR TITLE
companies crud

### PR DESCRIPTION
**Pull request by apotdevin**
# Details

- add: companies query
- add: companies add mutation
- add: companies delete mutation
- add: TextButton

# ToDo

- [ ] company subscription

# Notes

Missing the company update mutation. We should create first the update component (https://github.com/orgs/byteleaf/projects/1#card-52235451)

I think we should add a `confirmation` modal to confirm a user want's to delete a company. Don't think it's that huge of a deal right now but we can think about adding a modal component later :)

### CODE REVIEW

File: src/components/Atoms/Button/Actions/DeleteCompany/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  action: TableRowAction;
+  row: TableRow;
+  withMargin: boolean;
+};
+
+export default ({ action: { label }, row, withMargin }: DeleteButtonProps) => {
+  const [deleteCompany, { loading }] = useDeleteCompanyMutation({
+    refetchQueries: ['companies'],
+    onError: (err) => {
+      console.error(err);
```
Comment (apotdevin) [inline src/components/Atoms/Button/Actions/DeleteCompany/index.tsx:16]: Any preferred way we handle errors? Left a console log but need to clean this up

File: .eslintignore [hunk 2]
```diff
@@ -2,3 +2,6 @@
 /build
 babel-plugin-macros.config.js
 postcss.config.js
+
+**/__generated__/*.tsx
+src/graphql/types.ts
```
Comment (apotdevin) [inline .eslintignore:7]: Had to re add this, forgot the commit hooks also format these files

File: src/components/Atoms/Button/Actions/DeleteCompany/index.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import { TableRow, TableRowAction } from '../../../../Organisms/DataTable/types';
+import TextButton from '../../TextButton';
+
+export type DeleteButtonProps = {
+  action: TableRowAction;
+  row: TableRow;
+  withMargin: boolean;
+};
+
+export default ({ action: { label }, row, withMargin }: DeleteButtonProps) => {
```
Comment (heerma) [inline src/components/Atoms/Button/Actions/DeleteCompany/index.tsx:12]: We should rethink about how we structure our project, I would love to have atoms and molecules really generic (like a component library) without domain specific code. I know that it's already not like this, but that's why i would love to have a small meeting about this :)

File: src/components/Atoms/Button/Actions/DeleteCompany/index.tsx (hunk 1 above)
Comment (heerma) [reply src/components/Atoms/Button/Actions/DeleteCompany/index.tsx:16]: We need a strategy for this, but i think easiest would be a notification component that displays the error. Could be a follow up ticket

File: src/pages/companies.tsx [hunk 4]
```diff
@@ -41,24 +37,19 @@ const createEmptyTableConfig = (t: TFunction): TableConfig => ({
...
 
-  const [tableConfig] = React.useState<TableConfig>(createEmptyTableConfig(t));
-
-  const [tableData, setTableData] = React.useState<TableData>(emptyTableData);
+  const [tableConfig] = useState<TableConfig>(createEmptyTableConfig(t));
 
-  const { data } = useCompaniesQuery({ fetchPolicy: 'network-only' });
+  const { data } = useCompaniesQuery();
 
-  useEffect(() => {
```
Comment (markus-heer) [inline src/pages/companies.tsx:50]: Why was this removed? Shouldn't the mapping only be done on data change, not for every rerender?

File: src/components/Atoms/Button/Actions/DeleteCompany/index.tsx (hunk 3 above)
Comment (apotdevin) [reply src/components/Atoms/Button/Actions/DeleteCompany/index.tsx:12]: Definitely! Agree with this, didn't know where to put it but left it in a separate folder to move it easily

File: src/pages/companies.tsx (hunk 4 above)
Comment (apotdevin) [reply src/pages/companies.tsx:50]: Ah yes, imo I feel like this is an optimisation that put's in more complexity than what it brings. You need to add an additional state and keep it in sync with the data (that should already be handled by apollo's cache) to avoid a map that isn't very intensive.

What do you think? If you still prefer this optimisation maybe I can move it over to a `useMemo`?

Comment (markus-heer) [reply src/pages/companies.tsx:50]: Seems legit, let's go with your solution :)

**Review by markus-heer — APPROVED** [review]
